androidHyperskillApp/src/main/java/org/hyperskill/app/android/step_quiz_code/view/fragment/CodeStepQuizFragment.kt

```diff
@@ -138,8 +139,21 @@ class CodeStepQuizFragment :
             viewBinding = binding,
             codeLayoutDelegate = createCodeLayoutDelegate(),
             codeStepQuizConfig = config,
-            onFullscreenClicked = ::onFullScreenClicked,
-            onQuizChanged = ::syncReplyState
+            callback = object : CodeStepQuizFormDelegate.Callback {
+                override fun onCodeGenerationAlertClick() {
+                    stepQuizViewModel.onNewMessage(
+                        StepQuizFeature.Message.FixGptGeneratedCodeMistakesBadgeClickedQuestionMark
+                    )
```

Comment:
`DefaultStepQuizFragment` has function `logAnalyticEventMessage` for that cases. So you should decide will inheritors has direct access to the `stepQuizViewModel`.

Reply:
@ivan-magda 
In case we would have a special method for all the StepQuizFeature messages it would be too many methods.
So I decided to make viewModel protected.